java/core/src/java/org/apache/orc/impl/Dictionary.java

```diff
@@ -47,6 +47,15 @@
    */
   void getText(Text result, int position);
 
+  /**
+   * Given the position index, write the original string, before being encoded,
+   * to the OutputStream.
```

Comment:
Can we please mention here that this is directly writing to out without extra copies?

Reply:
@pgaref Thanks for checking this out too.

Since this is an API, I don't think there should be that level of details/constraints in regards to the implementations.  Anyone implementing this API can choose to do what they want. :)

Reply:
Thats true, but the way we pass the text argument here this seems to be an one way option -- wound not mind adding the explanation on getText HT/Tree implementations though